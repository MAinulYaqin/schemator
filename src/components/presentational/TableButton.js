import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import "typeface-roboto";

const Container = styled.div`
  background: #444;
  width: 100%;
  height: 32px;
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: 4px;
  cursor: pointer;
`;

const Caption = styled.p`
  color: white;
  font-size: 11px;
  font-family: Roboto;
  user-select: none;
`;

const TableButton = ({ caption, onClick }) => (
  <Container onClick={onClick}>
    <Caption>{caption}</Caption>
  </Container>
);

TableButton.propTypes = {
  caption: PropTypes.string,
  onClick: PropTypes.func
};

export default TableButton;
