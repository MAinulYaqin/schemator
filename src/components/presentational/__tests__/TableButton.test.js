import React from "react";
import { shallow } from "enzyme";

import TableButton from "../TableButton";

describe("TableButton", () => {
  const setup = propOverrides => {
    const props = {
      caption: "Add New Field",
      onClick: jest.fn(),
      ...propOverrides
    };

    const wrapper = shallow(<TableButton {...props} />);

    return {
      wrapper,
      props
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it("should calls 'onClick' while clicked", () => {
    const { wrapper, props } = setup();

    wrapper.simulate("click");

    expect(props.onClick).toBeCalled();
  });
});
