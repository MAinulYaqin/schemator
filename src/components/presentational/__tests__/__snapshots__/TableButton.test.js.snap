// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableButton should renders default 1`] = `
<styled.div
  onClick={[Function]}
>
  <styled.p>
    Add New Field
  </styled.p>
</styled.div>
`;
